load.go

```diff
@@ -86,12 +79,146 @@ func loadConfig(cfg Config, flags flagRefs, configurations ...any) error {
 	return nil
 }
 
+// readConfigurationFiles reads all configurations, appending slice values
+func readConfigurationFiles(cfg Config, files []string) (v *viper.Viper, err error) {
+	v = newViper(cfg)
+
+	for _, f := range files {
+		newV := newViper(cfg)
+
+		newV.SetConfigFile(f)
+		err = newV.ReadInConfig()
+		if err != nil {
+			if isNotFoundErr(err) {
+				cfg.Logger.Debug("no config file found, using defaults")
+			} else {
+				return nil, fmt.Errorf("unable to load config: %w", err)
+			}
+		}
+
+		all := v.AllSettings()
+		incoming := newV.AllSettings()
+
+		// merge configuration slices in priority order, so slices will have high priority entries first, and retain
+		// existing entries instead of overwriting them
+		err = mergo.Merge(&all, incoming, mergo.WithAppendSlice)
+		if err != nil {
+			return nil, err
+		}
+
+		// viper merge will overwrite same keys, we have appended slices to the previous config in the previous step
+		err = v.MergeConfigMap(all)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// we had been setting config previously to a string, so keep this behavior for now;
+	// viper seems to magically split this string if the target is a []string
+	v.Set("config", strings.Join(files, ","))
+
+	return v, nil
+}
+
+// newViper returns a configured new viper instance
+func newViper(cfg Config) *viper.Viper {
+	// EnvKeyReplacer allows for nested options to be specified via environment variables
+	// e.g. pod.context = APPNAME_POD_CONTEXT
+	var v = viper.NewWithOptions(viper.EnvKeyReplacer(strings.NewReplacer(".", "_", "-", "_")))
+
+	// load environment variables
+	v.SetEnvPrefix(cfg.AppName)
+	v.AllowEmptyEnv(true)
+	v.AutomaticEnv()
+	return v
+}
+
+// findConfigurationFiles returns the set of configuration files to use, either directly configured
+// or found in search paths, returning files in precedence order
+func findConfigurationFiles(cfg Config) (files []string, err error) {
```

Comment:
nit: order wise this reads easier when placing before `readConfiguration` ( so the order of execution follows the order of read)